internal/core/src/common/Schema.h

```diff
@@ -100,50 +97,43 @@ class Schema {
     }
 
     const FieldMeta&
-    operator[](FieldOffset field_offset) const {
-        Assert(field_offset.get() >= 0);
-        Assert(field_offset.get() < fields_.size());
-        return fields_[field_offset.get()];
+    operator[](FieldId field_id) const {
+        Assert(field_id.get() >= 0);
+        AssertInfo(fields_.count(field_id), "Cannot find field_id");
+        return fields_.at(field_id);
     }
 
     auto
     get_total_sizeof() const {
         return total_sizeof_;
     }
 
-    const std::vector<int64_t>&
-    get_sizeof_infos() const {
-        return sizeof_infos_;
-    }
-
-    FieldOffset
-    get_offset(const FieldName& field_name) const {
-        Assert(name_offsets_.count(field_name));
-        return name_offsets_.at(field_name);
-    }
-
-    FieldOffset
-    get_offset(const FieldId& field_id) const {
-        Assert(id_offsets_.count(field_id));
-        return id_offsets_.at(field_id);
+    FieldId
+    get_field_id(const FieldName& field_name) const {
+        AssertInfo(name_ids_.count(field_name), "Cannot find field_name");
+        return name_ids_.at(field_name);
     }
 
-    const std::vector<FieldMeta>&
+    const std::unordered_map<FieldId, FieldMeta>&
```

Comment:
it's better to move this change to another PR

Reply:
This pr is relatively large, and it is too troublesome to split out another pr now.